### ISSUE TITLE
Better way of connecting components, to allow sanity checks and more

### ISSUE THREAD

**Issue by samuell** [scipipe/scipipe#26]
If we create special `InPort` and `OutPort` structs, with some convenience functionality, we can move from:

```go
task2.InPorts["bar"] = task1.OutPorts["foo"]
```

... to something like:

```go
task2.InPorts["bar"].connectFrom(task1.OutPorts["foo"])
```

... and one could allow going the other direction too:

```go
task1.OutPorts["foo"].connectTo(task2.InPorts["bar"])
```

This should also work well with static port fields, such as:

```go
task2.InBar.connectFrom(task1.OutFoo)
task1.OutFoo.connectTo(task2.InBar)
```

This would allow us to make sure that there are no unconnected ports and other sanity checks, as well as to enable traversing the workflow dependency graph to produce a textual or graphical representation of the workflow.

An alternative approach would be to create a `Channel` component, that the "port" maps are initialized with, so that the assignment syntax still works (just that it is a Channel struct (with a real channel inside) that is assigned rather than a plain channel), but that wouldn't allow us the benefits stated above.

**Comment by samuell** [comment]
Some alternative syntaxes:

```go
task2.InBar.pipeFrom(task1.OutFoo)
task1.OutFoo.pipeTo(task2.InBar)
```

```go
task2.InBar.pipeFrom(task1.OutFoo)
task1.OutFoo.pipeInto(task2.InBar)
```

```go
task2.InBar.connect(task1.OutFoo)
task1.OutFoo.connect(task2.InBar)
```

```go
task2.InBar.pipe(task1.OutFoo)
task1.OutFoo.pipe(task2.InBar)
```

```go
task2.InBar.from(task1.OutFoo)
task1.OutFoo.to(task2.InBar)
```

```go
task2.InBar.from(task1.OutFoo)
task1.OutFoo.into(task2.InBar)
```

**Comment by samuell** [comment]
Well, voting is now:
```
1: -
2. -
3: -
4: 3
5: 2
6: 3
7: -
```

**Comment by samuell** [comment]
Implemented as of 8e73cab5bc0ea353bc130cde1fcee4c724c1e96f
